ols/src/llms/providers/azure_openai.py

```diff
@@ -52,6 +54,47 @@ def default_params(self) -> dict[str, Any]:
             "verbose": False,
         }
 
+        if self.credentials is not None:
+            # if credentials with API key is set, use it to call Azure OpenAI endpoints
+            default_parameters["api_key"] = self.credentials
+        else:
+            # credentials for API key is not set
+            # -> it is needed to retrieve token first
+            access_token = self.retrieve_access_token(azure_config)
+            default_parameters["azure_ad"] = access_token.token
```

Comment:
What `azure_ad` means?

Reply:
I've updated PR with better info (and realized there are two such parameters, switched to longer one)